pyomo/contrib/pyros/util.py

```diff
@@ -459,6 +657,11 @@ def partition_powers(n, v):
         starts = (0,) + starts + (n+v,)
         yield [starts[i+1] - starts[i] - 1 for i in range(v)]
 
+def sort_partitioned_powers(powers_list):
+    powers_list = sorted(powers_list, reverse=True)
+    powers_list = sorted(powers_list, key=lambda elem: max(elem))
```

Comment:
Out of curiosity, why sort this list twice?

Reply:
This sorting is to achieve the same ordering of powers as is generated by itertools.combinations, because I need to know the decision rule coefficient corresponding to each term for when we do the polishing step. The sorts are first lexicographic and then by max power.